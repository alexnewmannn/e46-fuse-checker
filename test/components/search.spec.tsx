import React from 'react';
import { axe } from 'jest-axe';
import Search from '../../components/search';
import { render } from '../test-utils';

jest.mock('../../data/fuses.json');

describe('Search', () => {
  it('should match the snapshots', () => {
    const { asFragment, debug } = render(<Search />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have no accessibility violations', async () => {
    const { container } = render(<Search />, {});
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
