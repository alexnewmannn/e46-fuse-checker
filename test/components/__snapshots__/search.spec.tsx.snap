// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshots 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="search__Form-sc-bhhmub-0 hDbVJe"
  >
    <div
      class="search__FormGroup-sc-bhhmub-1 bDbbVQ"
    >
      <label
        class="search__Label-sc-bhhmub-4 hfgjNB"
        for="search"
      >
        Search for a fuse
      </label>
      <input
        class="search__Input-sc-bhhmub-3 edbpFA"
        id="search"
      />
    </div>
  </form>
</DocumentFragment>
`;
